### PR TITLE
fix: paginate issues

### LINKED ISSUE #80: Fetch all devpool issues
The public and private issue fetch calls do not leverage `octokit.paginate` which is contributing to the current task issue onsite.

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves https://github.com/ubiquity/work.ubq.fi/issues/80

### CODE REVIEW

File: src/home/rendering/render-github-issues.ts [hunk 1]
```diff
@@ -17,8 +17,14 @@ export function renderGitHubIssues(tasks: TaskMaybeFull[]) {
...
   const baseDelay = 1000 / 15; // Base delay in milliseconds
-
   for (const task of tasks) {
+    const isBot = task.preview.user?.type === "Bot"
+    const userId = task.preview.user?.id
+
+    if (isBot && userId !== 113181824 && userId !== 76412717 && userId !== 165700353 && userId !== 133917611) {
+      continue;
+    }
+
```
Comment (0x4007) [inline src/home/rendering/render-github-issues.ts]: Quick fix but we should remember to implement correctly asap.

File: src/home/rendering/render-github-issues.ts [hunk 2]
```diff
@@ -75,11 +81,10 @@ function setUpIssueElement(
...
   issueElement.innerHTML = `
-      <div class="info"><div class="title"><h3>${
-        task.preview.title
-      }</h3></div><div class="partner"><p class="organization-name">${organizationName}</p><p class="repository-name">${repositoryName}</p></div></div><div class="labels">${labels.join(
-        ""
-      )}${image}</div>`;
+      <div class="info"><div class="title"><h3>${task.preview.title
+    }</h3></div><div class="partner"><p class="organization-name">${organizationName}</p><p class="repository-name">${repositoryName}</p></div></div><div class="labels">${labels.join(
+      ""
+    )}${image}</div>`;
```
Comment (0x4007) [inline src/home/rendering/render-github-issues.ts]: What is this? Can you run the formatter?

Comment (Keyrxng) [reply src/home/rendering/render-github-issues.ts]: format-on-save I believe, no actual changes were made here, I'll run format.

File: src/home/rendering/render-github-issues.ts (hunk 1 above)
Comment (Keyrxng) [reply src/home/rendering/render-github-issues.ts]: I took the 3 IDs at the end from your [comment](https://github.com/ubiquity/devpool-directory-bounties/issues/30#issuecomment-2292647684)

I see I see my mistake

Comment (Keyrxng) [reply src/home/rendering/render-github-issues.ts]: This error while implementing things as suggested

> You must authenticate with an access token authorized to a GitHub App in order to list installations - https://docs.github.com/rest/apps/installations#list-app-installations-accessible-to-the-user-access-token

So we'd need to store our bot private key here so we can verify that way you proposed for the cron job? 

I think it's overkill for something that will be removed in no time at all once the cron is completed.

Comment (rndquu) [reply src/home/rendering/render-github-issues.ts]: Is this logic correct? I mean that if an issue in https://github.com/ubiquity/devpool-directory is created by a user then `isBot = false` and the `if (isBot && userId !== 113181824 && userId !== 76412717 && userId !== 165700353 && userId !== 133917611)` expression evaluates to `false` thus rendering an issue by "unauthorized" user. While the expected behavior is to skip rendering such github issues.

Overall it should be simply filtered on the backend instead in order to omit such dirty hacks.

Comment (rndquu) [reply src/home/rendering/render-github-issues.ts]: I've removed all issues created by `testbot6` in https://github.com/ubiquity/devpool-directory/issues so this check seems to be redundant:
```
    if (isBot && userId !== 113181824 && userId !== 76412717 && userId !== 165700353 && userId !== 133917611) {
      continue;
    }
```

Again, it's better to simply fix it on the backend https://github.com/ubiquity/devpool-directory-bounties/issues/31

Comment (Keyrxng) [reply src/home/rendering/render-github-issues.ts]: > Overall it should be simply filtered on the backend instead in order to omit such dirty hacks.

Agreed.

> Is this logic correct? 

In terms of preventing bots yeah but I hadn't considered that user's can create issues too

> Again, it's better to simply fix it on the backend https://github.com/ubiquity/devpool-directory-bounties/issues/31

So I should remove these checks here and we address the backend immediately instead

**Comment by rndquu** [comment]
Since there are no more "spam" github issues created by `testbot6` at https://github.com/ubiquity/devpool-directory/issues then it makes sense to 
1. Keep this PR responsible only for the pagination feature 
2. Remove [bot checks](https://github.com/ubiquity/work.ubq.fi/pull/81/files#diff-dd804fd19e44d553ea26928f9e72a6198775a1c3af9605afebd7935c5b57d05aR21-R27) and let the backend [handle](https://github.com/ubiquity/devpool-directory-bounties/issues/31) it

**Comment by Keyrxng** [comment]
cleaned up the commit history a bit and this PR now only adds pagination

**Review by rndquu — APPROVED** [review]

**Pull request merged**